### ISSUE TITLE
RIGHT JOIN with a predicate that compares two integer columns results in an "Unhandled type" error

### ISSUE THREAD

**Issue by mrigger** [cwida/duckdb#503]
Consider the following statements:

```sql
CREATE TABLE t0(c0 INT);
CREATE TABLE t1(c0 INT);
SELECT * FROM t0 RIGHT JOIN t1 ON t0.c0!=t1.c0; -- Error: Unhandled type for empty NL join
```
Unexpectedly, the `SELECT` results in an error `Error: Unhandled type for empty NL join`. The predicate `t0.c0=t1.c0` works as expected.

I found this bug on the latest master commit 4ad62e9e13f4a1b3c6452bcb4f5f31eac3d86053.

Here is the detailed output:
```
duckdb/build/release$ ./duckdb_cli 
SQLite version DuckDB 4ad62e9e
Enter ".help" for usage hints.
Connected to a transient in-memory database.
Use ".open FILENAME" to reopen on a persistent database.
sqlite> CREATE TABLE t0(c1 INT);
sqlite> CREATE TABLE t1(c1 INT);
sqlite> SELECT * FROM t0 RIGHT JOIN t1 ON t0.c1!=t1.c1;
Error: Unhandled type for empty NL join
```

**Comment by Mytherin** [comment]
Thanks for the report! Seems to be an issue with us not supporting RIGHT JOIN currently. RIGHT JOINs should simply be rewritten to a LEFT JOIN in the Transformer in my opinion.

**Comment by Mytherin** [comment]
Fixed this (along with #505) in 1014b562dcd8228e7bdb3f984323034d3f0ae3ca by writing RIGHT joins to LEFT joins. Previously, right joins were not supported but this was not properly checked for, as such weird behavior could happen.
